libraries/ArduinoOTA/ArduinoOTA.cpp

```diff
@@ -70,7 +84,12 @@ void ArduinoOTAClass::begin() {
     _port = 8266;
   }
 
-  _udp_ota.begin(_port);
+  _udp_ota = new UdpContext;
```

Comment:
You have to check that `_udp_ota` is nullptr before creating another one. If it isn't, `unref` it first. This could happen if someone calls begin() twice for some reason.

Also destructor is missing... Add it along with `if (_udp_ota) _udp_ota->unref();`. I know that ArduinoOTA is a static object now, but it's omission makes any c++ programmer weep :)